assume/markets/clearing_algorithms/redispatch.py

```diff
@@ -37,16 +38,25 @@ class RedispatchMarketRole(MarketRole):
 
     """
 
+    required_fields = ["node"]
+
     def __init__(self, marketconfig: MarketConfig):
         super().__init__(marketconfig)
-
         self.network = pypsa.Network()
         # set snapshots as list from the value marketconfig.producs.count converted to list
         self.network.snapshots = range(marketconfig.market_products[0].count)
-        self.solver = marketconfig.param_dict.get("solver", "glpk")
-        network_path = marketconfig.param_dict.get("network_path")
+        assert self.grid_data
+        # set backup marginal cost
+        backup_marginal_cost = marketconfig.param_dict.get("backup_marginal_cost", 10e4)
```

Comment:
sometimes we use self.market_config, sometimes just the market config. Should we use one version only?